### PR TITLE
[Elastic Agent] Fix issue with ensureServiceToken.

### LINKED ISSUE #29799: [Elastic Agent] ensureServiceToken should not always be called for the container
With the latest change from #29651 the Elastic Agent container is now always trying to create a service token. When just running the Elastic Agent container in standalone mode this should not happen.

For confirmed bugs, please report:
- Version: 8.1.0-SNAPSHOT
- Operating System: Docker
- Steps to Reproduce: 

Just start the container and see that it fails to contact Kibana. It should only do that if its going to bootstrap a Fleet Server:

`docker run -it docker.elastic.co/beats/elastic-agent:8.1.0-SNAPSHOT`

Logs the following error:

````
Requesting service_token from Kibana.
Error: request to get security token from Kibana failed: fail to execute the HTTP POST request: Post "http://kibana:5601/api/fleet/service-tokens": dial tcp 127.0.0.1:5601: connect: connection refused
For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.1/fleet-troubleshooting.html
```

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Fixes #29799.

It was always calling `ensureServiceToken` but that should only be done in the case that Fleet Server is enabled for the container.

## Why is it important?

Otherwise its not possible to start the container without it trying to communicate with Kibana.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## How to test this PR locally

Build the container:

```
$ PLATFORMS="linux/amd64" mage dev:package
$ docker run -it docker.elastic.co/beats/elastic-agent:8.1.0
```

## Related issues

- Closes #29799

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/cmd/container.go [hunk 1]
```diff
@@ -287,6 +282,19 @@ func runContainerCmd(streams *cli.IOStreams, cmd *cobra.Command, cfg setupConfig
...
 	}
 	if cfg.Fleet.Enroll {
+		if cfg.FleetServer.Enable {
+			if client == nil {
+				client, err = kibanaClient(cfg.Kibana, cfg.Kibana.Headers)
+				if err != nil {
+					return err
+				}
+			}
+			err = ensureServiceToken(streams, client, &cfg)
```
Comment (michel-laterman) [inline x-pack/elastic-agent/pkg/agent/cmd/container.go:292]: Shouldn't this move to the line 273? That way the call to `kibanaSetup` uses a `service_token` as well

**Comment by blakerouse** [comment]
@michel-laterman Your correct it would be better moved up. I also messed up the client change as well, add comment on why it needs to remain in the function.

**Review by michel-laterman — APPROVED** [review]
lgtm, thanks for making the change!

**Comment by blakerouse** [comment]
@adam-stokes Another Ansible failure, just so you know.

All tests pass, other than the failed ansible job.

**Pull request merged**